documentation/docs/payid.mdx

```diff
@@ -0,0 +1,74 @@
+---
+id: payid
+title: PayID
+---
+
+import { Mermaid } from './Mermaid'
+
+By default, Open Payments interactions centre around an [account](./accounts)
+resource and all APIs are relative to the account URL.
+
+The account URL is [discovered](./discovery) through a
+[Payment Pointer](./terminology#payment-pointer) which is an account identifer
+for Interledger-enabled accounts.
+
+Said differently, any account identified by a Payment Pointer can be paid into
+using the [STREAM](terminology#STREAM) protocol.
+
+This feature allows clients setup payments, for uses cases such
+[web monetization](./web-monetization), with a single API call, to get the
+payment details, and be guaranteed that the receiving account supports ILP as
+payment method.
+
+A **PayID** is an alternative identifier which, unlike Payment Pointers that
+identify accounts, identifies [account owners](./terminology#account-owner).
+
+The
+[PayID URI specification](https://github.com/xpring-eng/rfcs/blob/master/payid/dist/spec/payid-uri.txt)
+defines the format of a PayID to be `acctpart "$" host`.
+
+With respect to Open Payments the `acctpart` is the account owner identifier and
+`host` is the origin of the Open Payments server.
+
+The following PayID identifies an account for the owner `alice` at the wallet
+`wallet.example`:
+
+> `alice$wallet.example`
+
+## Open Payments PayID Discovery
+
+As with all APIs in Open Payments the [owner](./owner.mdx) is a resource and is
+identified by a URL.
+
+The
+[PayID URI specification](https://github.com/xpring-eng/rfcs/blob/master/payid/dist/spec/payid-uri.txt)
+allows for any protocol to define a discovery protocol on top of the identifier.
+For Open Payments the discovery protocol is as follows:
+
+1. Given a PayID conforming to the prescribed format, `acctpart "$" host`,
+   extract the values **acctpart** and **host**.
+2. Let **path** be the string `/$/` concatenated with **acctpart**
+3. Let **account owner URL** be a new URL as defined in
+   [WHATWG URL](https://url.spec.whatwg.org/#concept-url) where:
+   1. The URL's scheme is `https`
+   2. The URL's host is **host**
+   3. The URL's path is **path**
+
+For example, the PayID `alice$wallet.example` resolves to the URL
```

Comment:
Let's debate/discuss this a bit more. While I do like the simplicity, I don't like stepping on an operator's path-space.

So, it would be more ideal if we had a very simple well-known location for openpayments with a simple template that would allow operators to define this path.

I have not been persuaded that the Webfinger approach is "bad" or "too hard" or whatever -- in fact, I have a very simple template syntax, and supporting webfinger can be done with a single static file in the simplest of use-cases, all while providing full domain-customization.

In fact, I would argue that supporting a 2-line JRD in a static file is about as much work from a programming perspective as supporting this manual assembly. So, why not just use the solution that gives domain operators more power?

Reply:
One more thing - I think it will become necessary to support arbitrary meta-data for a PayID, which Webfinger supports very nicely (just add another 2-line JRD to an array).

For example, we'll likely want to define with a bit more fidelity some kind of "Wallet Discovery" protocol that allows software using OpenPayments to redirect a user to a wallet-link in order to make a payment.

Another example is any yet-to-be defined protocols that might need to facilitate PayID-level metadata.

Reply:
> I don't like stepping on an operator's path-space.

Why? What are the odds of a path collision?
If there is a collision it's simple to get around by simply using a subdomain.

You are putting technical purity at the top of the order of constituents and it should be at the bottom

Reply:
>  I think it will become necessary to support arbitrary meta-data for a PayID

In Open Payments that is done after discovery. Once the URL of the account owner is known we can add whatever additional meta-data we need to the owner resource.

Reply:
> Why? What are the odds of a path collision?

Path-collision isn't the only concern. What about operators that don't have fine-grained control over their domain-space (e.g., it's possible that just "adding a sub-domain" may not not work for certain operators).

Reply:
> You are putting technical purity at the top of the order of constituents and it should be at the bottom

This feels pejorative -- the comment doesn't actually make a technical case for why "robust discovery" should be at the bottom of the order as opposed to the top.

In my view, we're building a decentralized protocol here with open-payments. We need to be thoughtful about the restrictions/assumptions we make about it because if it ever becomes widely deployed, we won't be able to easily take these decisions back.

Overall, whether we use PayID Discovery or not here, I do feel like the `./well-known` space is the more appropriate location for open-payments meta-data and discovery. Perhaps we go back to the original OpenPayments Discovery endpoints? That location could still be easily derived from a PayID (whether using PayID Discovery or not) and we'll have a lot more flexibility in terms of what we put into `https://{host}/.well-known/open-payments`.

What do you think about that?

Reply:
I think this is a mis-use of the well-known address space which to my understanding is for hosting services per host. 

A PayID has two parts, the `host` and `account` so transposing it to a URL in the well-known space would mean something like:

`https://{host}/.well-known/payid/{account}`

Also, where I think a PayID is unique is that it's a human-friendly identifier and there is value in the URL form being human-friendly too.

I'm hopeful that wallets will actually host `text/html` representations of their owner resources that are something like an open protocol version of [Cash Tags](https://cash.app/help/us/en-gb/3123-cashtags)

As a wallet I'd much prefer my users to have a public page at `https://wallet.example/$/david` that is a direct transposition of their PayID than `https://wallet.example/.well-known/payid/david`